NimbleMedium/Sources/Presentation/Modules/Feeds/FeedsTab/FeedsTabView.swift

```diff
@@ -0,0 +1,129 @@
+//
+//  FeedsTabView.swift
+//  NimbleMedium
+//
+//  Created by Mark G on 11/08/2021.
+//
+
+import Refresh
+import Resolver
+import SwiftUI
+import ToastUI
+
+struct FeedsTabView: View {
+
+    @ObservedViewModel private var viewModel: FeedsTabViewModelProtocol = Resolver.resolve()
+    
+    @State private var isFirstLoad: Bool = true
+    @State private var isErrorToastPresented = false
+    @State private var isShowingCreateArticleScreen = false
+
+    var body: some View {
+        Group {
+            if isFirstLoad {
+                ProgressView()
+            } else {
+                FeedList(viewModel: viewModel)
+            }
+        }
+        .frame(maxWidth: .infinity, maxHeight: .infinity)
+        .toast(isPresented: $isErrorToastPresented, dismissAfter: 3.0) {
+            ToastView(Localizable.errorGeneric()) { } background: { Color.clear }
+        }
+        .onReceive(viewModel.output.didFinishRefresh) { _ in if isFirstLoad { isFirstLoad = false } }
+        .onReceive(viewModel.output.didFailToLoadArticle) { _ in isErrorToastPresented = true }
+        .fullScreenCover(isPresented: $isShowingCreateArticleScreen) { CreateArticleView() }
+        .onAppear {
+            viewModel.input.refresh()
+        }
+    }
+}
+
+// MARK: FeedList
+private extension FeedsTabView {
```

Comment:
You should put `private` with the truct
```suggestion
extension FeedsTabView {
```

```swift
    private struct FeedList: View, Equatable
```